schema/spacewalk/common/tables/suseISSHub.sql

```diff
@@ -0,0 +1,25 @@
+--
+-- Copyright (c) 2024 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+CREATE TABLE suseISSHub
+(
+    id 	                NUMERIC NOT NULL
```

Comment:
```suggestion
    id 	                BIGINT PRIMARY KEY GENERATED ALWAYS AS IDENTITY
```

Please no NUMERIC.
Using this you can also drop custom sequence and also will not require insert to have `sequence_nextval(...)`, on contrary it will forbid custom values there so there is no risk of accidental sequence bug.

Hibernate will need to have `@GeneratedValue(strategy=GenerationType.IDENTITY)` for this to work AFAIK, I did not test it yet.

Reply:
And same for all other `id` columns here.

Reply:
I agree with Ondrej, we should not use Numeric. In this case, I think an Integer is enough, since it can save a number up to 2147483647

Reply:
There is indeed a storage saving (4B vs 8B per row), but performance wise these two are very close to each other.
Given how we regularly copy paste existing implementations when adding new, I argue we should default to `BIGINT` for all `id` columns, regardless expected table size.

Reply:
we, I think it makes sense. Thank you for bring it Ondrej :)

Reply:
What about mixing? The foreign key to mirror_creds_id is also NUMERIC. Do we have to keep it as the ID column is defined that way?

Reply:
We can mix types in FK. (SQL standard AFAIK forbid it completely, pgsql allows something like SMALLINT->INTEGER, etc.)
So with FK to existing tables we will have to use what is already there.

Reply:
I think this is fixed now.